lib/chef-vault/mixins.rb

```diff
@@ -0,0 +1,36 @@
+class ChefVault
+  module Mixins
+    # In Chef 12, Chef Solo can have an array of data_bag paths, rather
+    # than just a string. To cope with that, we'll:
+    #  1. Look for an existing data bag item in any of the configured
+    #     paths and use that by preference
+    #  1. Otherwise, just use the first location in the array
+    def find_solo_path(item_id)
+      if Chef::Config[:data_bag_path].kind_of?(Array)
+        path = Chef::Config[:data_bag_path].select { |dir|
```

Comment:
Minor style suggestion: I like to use the `find` method when I just need the first item that matches, saving you the call #first.